shared/views/RoomDirectoryView.js

```diff
@@ -17,13 +20,39 @@ class RoomDirectoryView extends TemplateView {
       vm.setSearchTerm(searchInputValueBeforeRendering);
     }
 
+    function toggleSafeSearchLabel() {
+      var text = document.getElementById('checkBox_label');
+      if (text.innerHTML === 'Safe Search is on') {
+        text.innerHTML = 'Safe Search is off';
+      } else {
+        text.innerHTML = 'Safe Search is on';
+      }
+    }
```

Comment:
Instead of manipulating the DOM directly, we should update the view model and have Hydrogen re-render for us.

As an example see:

 - [`shared/views/DeveloperOptionsContentView.js#L26`](https://github.com/matrix-org/matrix-public-archive/blob/f95086df0d22393376916e087bc0ced09ba3f389/shared/views/DeveloperOptionsContentView.js#L26)
  - [`shared/viewmodels/DeveloperOptionsContentViewModel.js#L32-L39`](https://github.com/matrix-org/matrix-public-archive/blob/f95086df0d22393376916e087bc0ced09ba3f389/shared/viewmodels/DeveloperOptionsContentViewModel.js#L32-L39)